lxd/storage/drivers/utils_cgo.go

```diff
@@ -348,3 +360,25 @@ func UnsetAutoclearOnLoopDev(loopFd int) error {
 
 	return nil
 }
+
+func SetDirectIOLoopDev(loopFd int) error {
+	ret, err := C.set_direct_io_loop_device(C.int(loopFd))
```

Comment:
This surfaces the errno value, right?
So how about:
```go
if ret < 0 || err != nil {
        if err != nil {
               return err
        }

        return fmt.Errorf....
}

Reply:
I ended up removing both of the Go functions I added because the `set_direct_io_loop_device` C functions are only ever called from C code.